api-resources.md

```diff
@@ -0,0 +1,414 @@
+
+#### Representation
+
+A paginated list of aggregation jobs for the task.
+
+```
+struct {
+  u64 page;
+  AggregateJob aggregate_jobs<0..2^16-1>
+} AggregationJobList
+```
+
+#### Required HTTP methods
+
+##### GET/HEAD
+
+The request takes an optional query parameter for the page. The response body is a `struct AggregationJobList`.
+
+### An aggregate share
+
+Only the helper supports this resource.
+
+#### Path
+
+`/tasks/task-id/aggregate_shares/aggregate-share-id`
+
+`aggregate-share-id` uniquely identifies an aggregate share. It is chosen by the leader when it requests an aggregate share by the helper.
+
+#### Representation
+
+```
+struct {
+  HpkeCiphertext encrypted_aggregate_share;
+} AggregateShare;
+```
+
+#### Required HTTP methods
+
+##### GET/HEAD
+
+If the aggregate share is available, the result is a `struct AggregateShare`. Otherwise the server can return something like HTTP 202 Accepted to indicate it's not ready yet, or HTTP 404 if no such aggregate share is known to the helper.
+
+##### PUT
+
+PUT to an aggregate share resource is how the leader instructs the helper to compute aggregate shares. The request is an idempotent PUT, because if an aggregate share with the provided ID already exists, its representation can be returned by the helper without further state mutation.
+
+The body of the request is an `AggregateShareReq`:
+
+```
+struct {
+  QueryType query_type;
+  select (BatchSelector.query_type) {
+    case time_interval: Interval batch_interval;
+    case fixed_size: Empty;
+  };
+} BatchSelector;
+
+struct {
+  BatchSelector batch_selector;
+  opaque agg_param<0..2^16-1>;
+  uint64 report_count;
+  opaque checksum[32];
+} AggregateShareReq;
+```
+
+If successful, the response from the helper is a `struct AggregateShare`. That same share may later be obtained by a GET request on the resource.
+
+This is analogous to DAP-02's `POST /aggregate_share`.
+
+###### A note on `BatchSelector.fixed_size`
+
+In DAP-02, a `BatchSelector` with `query_type = fixed_size` contains a `BatchID`. In this API, the batch ID hoisted up into the resource URI. This implies that `time_interval`-type queries now also have a batch ID, chosen by the leader. While aggregate shares will often (always?) be 1:1 with collections, the IDs do not have to match.
+
+Note also that `AggregateShareReq` should use a `Query` (https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/issues/366).
+
+TODO: timg to read more about chunky DAP and think about how it intersects with this proposal, especially https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/issues/342
+
+##### DELETE
+
+Instructs the helper to abandon the aggregate share and allows the helper to discard all state related to it. This is akin to DELETE on a collection for the leader.
+
+### Aggregate shares
+
+Only the helper supports this resource.
+
+#### Path
+
+`/tasks/task-id/aggregate_shares`
+
+#### Representation
+
+A paginated list of aggregation jobs for the task.
```

Comment:
Why do we need a paginated list of aggregate shares?

I think this can be dropped from the specification (& left up to implementations, if they find it handy).

Reply:
Agreed, I will delete this resource.